tests/unit/query_v2_expression_evaluator.cpp

```diff
@@ -0,0 +1,1786 @@
+  auto *extract = EXTRACT("x", LITERAL("bla"), ADD(ident_x, LITERAL(1)));
+  const auto x_sym = symbol_table.CreateSymbol("x", true);
+  extract->identifier_->MapTo(x_sym);
+  ident_x->MapTo(x_sym);
+  EXPECT_THROW(Eval(extract), ExpressionRuntimeException);
+}
+
+TEST_F(ExpressionEvaluatorTest, RegexMatchInvalidArguments) {
+  EXPECT_TRUE(Eval(storage.Create<RegexMatch>(LITERAL(TypedValue()), LITERAL("regex"))).IsNull());
+  EXPECT_TRUE(Eval(storage.Create<RegexMatch>(LITERAL(3), LITERAL("regex"))).IsNull());
+  EXPECT_TRUE(Eval(storage.Create<RegexMatch>(LIST(LITERAL("string")), LITERAL("regex"))).IsNull());
+  EXPECT_TRUE(Eval(storage.Create<RegexMatch>(LITERAL("string"), LITERAL(TypedValue()))).IsNull());
+  EXPECT_THROW(Eval(storage.Create<RegexMatch>(LITERAL("string"), LITERAL(42))), ExpressionRuntimeException);
+  EXPECT_THROW(Eval(storage.Create<RegexMatch>(LITERAL("string"), LIST(LITERAL("regex")))), ExpressionRuntimeException);
+}
+
+TEST_F(ExpressionEvaluatorTest, RegexMatchInvalidRegex) {
+  EXPECT_THROW(Eval(storage.Create<RegexMatch>(LITERAL("text"), LITERAL("*ext"))), ExpressionRuntimeException);
+  EXPECT_THROW(Eval(storage.Create<RegexMatch>(LITERAL("text"), LITERAL("[ext"))), ExpressionRuntimeException);
+}
+
+TEST_F(ExpressionEvaluatorTest, RegexMatch) {
+  EXPECT_FALSE(Eval(storage.Create<RegexMatch>(LITERAL("text"), LITERAL(".*ex"))).ValueBool());
+  EXPECT_TRUE(Eval(storage.Create<RegexMatch>(LITERAL("text"), LITERAL(".*ext"))).ValueBool());
+  EXPECT_FALSE(Eval(storage.Create<RegexMatch>(LITERAL("text"), LITERAL("[ext]"))).ValueBool());
+  EXPECT_TRUE(Eval(storage.Create<RegexMatch>(LITERAL("text"), LITERAL(".+[ext]"))).ValueBool());
+}
+
+class ExpressionEvaluatorPropertyLookup : public ExpressionEvaluatorTest {
+ protected:
+  std::pair<std::string, memgraph::storage::v3::PropertyId> prop_age =
+      std::make_pair("age", shard_manager->NameToProperty("age"));
+  std::pair<std::string, memgraph::storage::v3::PropertyId> prop_height =
+      std::make_pair("height", shard_manager->NameToProperty("height"));
+  Identifier *identifier = storage.Create<Identifier>("element");
+  Symbol symbol = symbol_table.CreateSymbol("element", true);
+
+  void SetUp() { identifier->MapTo(symbol); }
+
+  auto Value(std::pair<std::string, memgraph::storage::v3::PropertyId> property) {
+    auto *op = storage.Create<PropertyLookup>(identifier, storage.GetPropertyIx(property.first));
+    return Eval(op);
+  }
+};
+
+// TODO(kostasrim) These will fail because of memory resource not propagating correctly. This should be done as part of
+// polishing the allocators.
+// TEST_F(ExpressionEvaluatorPropertyLookup, Vertex) {
+//   auto v1 = CreateVertex({{prop_age.second, memgraph::msgs::Value(static_cast<int64_t>(32))}}, shard_manager.get());
+//   frame[symbol] = TypedValue(v1);
+//   EXPECT_EQ(Value(prop_age).ValueInt(), 10);
+//   EXPECT_TRUE(Value(prop_height).IsNull());
+// }
+//  TEST_F(ExpressionEvaluatorPropertyLookup, Edge) {}
```

Comment:
Could you please create a task from this just to make sure we handle the allocators properly?